### PR TITLE
[primTorch] Add a ref for `addcmul`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #86731

Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Note: I've made a tiny change to the `OpInfo` to keep this self-contained. LMK if you want all the suggested changes (C++ and error inputs) from https://github.com/pytorch/pytorch/pull/72949 as part of this PR.

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -1582,6 +1583,36 @@ def addcdiv(
...
+    if value is not None:
+        dtype = self.dtype  # no scalars allowed, see add
+        python_type = utils.dtype_to_type(dtype)
+        if not utils.is_weakly_lesser_type(type(value), python_type):
+            msg = (
+                "value argument of type {0} cannot be safely cast to type {1}!".format(
+                    type(value), python_type
+                )
+            )
+            raise ValueError(msg)
```
Comment (lezcano) [inline torch/_refs/__init__.py:1611]: Better to use a `utils.check`.

**Review by lezcano — APPROVED** [review]
One small comment, otherwise LGTM!

**Review by mruberry — APPROVED** [review]
See @lezcano's note about using `utils.check()`, otherwise LGTM!

**Comment by nkaretnikov** [comment]
`addcmul`  failures from  https://github.com/pytorch/pytorch/issues/76047 no longer happen

**Pull request closed without merging**